Source/TractUI.cpp

```diff
@@ -67,25 +68,31 @@ void TractUI::mouseDrag(const juce::MouseEvent &e)
 	if(isTongue){
 		this->setTongue(props, index, diameter);
 	} else {
-		this->setConstriction(props, index, diameter);
+		if(processor.constrictionActive)
+			this->setConstriction(props, index, diameter);
 	}
 }
 
 void TractUI::mouseUp(const juce::MouseEvent &e)
 {
+	double index;
+	double diameter;
 	t_tractProps *props = this->processor.getTractProps();
+	double x = e.getMouseDownX() + e.getDistanceFromDragStartX() - this->originX;
+	double y = e.getMouseDownY() + e.getDistanceFromDragStartY() - this->originY;
+	
+	this->getEventPosition(props, x, y, index, diameter);
+	
 	if(!isTongue){
-		this->setConstriction(props, 0, 0);  //remove constriction
+		if(processor.constrictionActive)
+			this->setConstriction(props, index, processor.constrictionMin);  //remove constriction
 	}
 }
 
 void TractUI::setConstriction(t_tractProps *props, double index, double diameter)
 {
-	double constrictionMin = -2.0;
-	double constrictionMax = 2.0;
-	
 	this->processor.constrictionX = index/props->n;
-	this->processor.constrictionY = (diameter - constrictionMin)/(constrictionMax - constrictionMin);
+	this->processor.constrictionY = (diameter - processor.constrictionMin)/(processor.constrictionMax - processor.constrictionMin);
```

Comment:
Small thing, either do `this->processor` or `processor`, but certainly not both.